guides/set-current-user.md

```diff
@@ -0,0 +1,97 @@
+[Back to Main README](https://github.com/simplabs/ember-simple-auth/blob/master/README.md)
+
+## How to Set currentUser Data
+
+Although there are various ways to load currentUser data, this is the canonical
+way to do so. Your authentication route will need to identify how it can access
+the data for the current user. In this example, it is sent as the user_id on the
+authentication payload.
+
+### CurrentUser Service
+
+In order for the app to keep the user data, we will store it in its own service:
+
+```js
+// app/services/session-account.js
+import Ember from 'ember';
+
+const { inject: { service }, isEmpty, RSVP } = Ember;
+
+export default Ember.Service.extend({
+  session: service('session'),
+  store: service(),
+
+  account: null,
+
+  loadCurrentUser() {
+    return new RSVP.Promise((resolve, reject) => {
+      const userId = this.get('session.data.authenticated.user_id');
+      if (!isEmpty(userId)) {
+        return this.get('store').find('user', userId).then((user) => {
+          this.set('user', user);
+          resolve();
+        }, reject);
+      } else {
+        resolve();
+      }
+    });
+  }
+});
+```
+
+In this example, we grab the user id and then make an ember-data request to the
+API. After a successful request, it sets the user and is easily access anywhere
+by injecting this service.  Note: If no id is sent, then the user will not be
+set and this implementation will log you out.
+
+### Calling the function to set the current user
+
+The best place for us to make this call is on the application route:
+
+```js
+// app/routes/application.js
+import Ember from 'ember';
+import ApplicationRouteMixin from 'ember-simple-auth/mixins/application-route-mixin';
+
+const { service } = Ember.inject;
+
+export default Ember.Route.extend(ApplicationRouteMixin, {
+  sessionAccount: service(),
+
+  beforeModel() {
+    return this._loadCurrentUser();
+  },
+
+  sessionAuthenticated() {
+    this._super(...arguments);
+    this._loadCurrentUser().catch(() => this.get('session').invalidate());
+  },
+
+  _loadCurrentUser() {
+    return this.get('sessionAccount').loadCurrentUser();
+  }
+});
+```
+
+This uses the session-account service you created to set the currentUser when
+the application route loads and there is a session that has successfully been
+restored (using the beforeModel() hook), or when a session is authenticated
+(using the sessionAuthenticated() hook). Notice that in the
+sessionAuthenticated() hook, if we can't load the current user, then it will
+invalidate our session.
```

Comment:
This should also explain why it's necessary to call that method from 2 different places - `beforeModel` for that case that the application already starts up authenticated and `sessionAuthenticated` if the session becomes authenticated via the tab/windows syncing mechanism.